asmcomp/arm/emit.mlp

```diff
@@ -367,6 +367,34 @@ let emit_load_symbol_addr dst s =
     1
   end
 
+(* Emit instructions that set [rd] to 1 if integer condition [cmp] holds
+   and set [rd] to 0 otherwise. *)
+
+let emit_set_condition cmp rd =
+  let compthen = name_for_comparison cmp in
+  let compelse = name_for_comparison (negate_integer_comparison cmp) in
+  if !arch < ARMv8 || not !thumb then begin
+    `	ite	{emit_string compthen}\n`;
+    `	mov{emit_string	compthen}	{emit_reg rd}, #1\n`;
+    `	mov{emit_string compelse}	{emit_reg rd}, #0\n`;
+    3
+  end else begin
+    (* T32 mode in ARMv8 deprecates general ITE blocks
+       and favors IT blocks containing only one 16-bit instruction.
+       mov <reg>, #<imm> is 16 bits if <reg> is R0...R7 and <imm> is 0 or 1. *)
```

Comment:
I found this comment a bit misleading -- I know it says "if" not "iff", but it would probably be clearer if it said the actual condition on `<imm>`, which is that it fits in 8 bits.

Reply:
Right, I put the actual condition.